src/transport/http2.rs

```diff
@@ -289,15 +316,20 @@ fn spawn_recv_forwarder(req: Body, req_tx: Sender<hyper::Result<Bytes>>) -> Join
                     true
                 }
             };
-            if let Err(_cause) = req_tx.send_async(chunk).await {
-                // The receiver is gone, so we can't send any more data.
-                //
-                // This is a normal way for an interaction to end, when the server side is done processing
-                // the request and drops the receiver.
-                //
-                // don't log the cause. It does not contain any useful information.
-                trace!("Flume receiver dropped");
-                break;
+            // Drain all complete messages from the buffer, even in case of an error.
+            while let Some(msg) = try_get_length_prefixed(&buf) {
+                let item = bincode::deserialize(msg).map_err(RecvError::DeserializeError);
+                buf.drain(..4 + msg.len());
```

Comment:
would be nice if this was a ringbuffer or if bincode could do zero copy deserializjbg, a lot of allocations in this path :/

Reply:
It would not allocate once the buffer has the right size. Drain just moves stuff around. But you are right that this is inefficient.

In 99% of all cases the arriving Bytes is exactly the thing to be deserialised, so I was going to just try to deserialize directly from the Bytes if the buffer is empty.

Will add a todo.